File: app/src/main/java/com/getman/varnabeach/VolleyHelper.java
package com.getman.varnabeach;

import android.content.Context;

import com.android.volley.Request;
import com.android.volley.RequestQueue;
import com.android.volley.toolbox.Volley;

public class VolleyHelper {
    private static VolleyHelper instance;
    private RequestQueue queue;
    private static Context context;

    private VolleyHelper(Context ctx) {
        // in case context other than application one is provided
        context = ctx.getApplicationContext();
        getRequestQueue();
    }

    public static VolleyHelper getInstance(Context ctx) {
        if (instance == null) {
            instance = new VolleyHelper(ctx);
        }
        return instance;
    }

    public RequestQueue getRequestQueue(){
        if (queue == null) {
            queue = Volley.newRequestQueue(context);
        }
        return queue;
    }

    public <T> void addToRequestQueue(Request<T> request) {
        getRequestQueue().add(request);
    }
}

File: app/src/main/java/com/getman/varnabeach/data/VolleyHelper.kt
package com.getman.varnabeach.data

import android.content.Context
import com.android.volley.Request
import com.android.volley.RequestQueue
import com.android.volley.toolbox.Volley
import dagger.hilt.android.qualifiers.ApplicationContext
import javax.inject.Inject

class VolleyHelper @Inject constructor(@ApplicationContext private val context: Context) {
    private var queue: RequestQueue? = null
    private val requestQueue: RequestQueue
        get() {
            if (queue == null) {
                queue = Volley.newRequestQueue(context)
            }
            return queue as RequestQueue
        }

    fun <T> addToRequestQueue(request: Request<T>?) {
        requestQueue.add(request)
    }
}